ansible_collections/arista/avd/plugins/action/yaml_templates_to_facts.py

```diff
@@ -120,22 +130,25 @@ def run(self, tmp=None, task_vars=None):
                 debug_item['timestamps']['done'] = datetime.now()
                 avd_yaml_templates_to_facts_debug.append(debug_item)
 
-        if output_filename:
+        # If the argument 'dest' is set, write the output data to a file.
+        if dest:
             if debug:
                 debug_item = {
-                    'action': 'output_filename',
-                    'output_filename': output_filename,
-                    'output_format': output_format,
+                    'action': 'dest',
+                    'dest': dest,
                     'timestamps': {
                         'write_file': datetime.now()
                     }
                 }
 
-            if output_format in ["yml", "yaml"] :
-                write_file_result = self.write_file(yaml.dump(output, indent=2, sort_keys=False, width=130), output_filename, task_vars)
+            # Depending on the file suffix of 'dest' (default: 'json') we will format the data to yaml or just write the output data directly.
+            # The Copy module used in 'write_file' will convert the output data to json automatically.
+            if dest.split('.')[-1] or '' in ["yml", "yaml"] :
```

Comment:
Why do we need ```or ''```

FYI
![image](https://user-images.githubusercontent.com/88734745/151830595-f12a7c48-1f8e-4c37-9093-066fc35782f7.png)